lldb/test/API/functionalities/progress_reporting/swift_progress_reporting/TestSwiftProgressReporting.py

```diff
@@ -0,0 +1,67 @@
+"""
+Test that we are able to broadcast and receive swift progress events from lldb
+"""
+import lldb
+from lldbsuite.test.lldbtest import *
+from lldbsuite.test.decorators import *
+import lldbsuite.test.lldbutil as lldbutil
+import threading
+
+class TestSwiftProgressReporting(TestBase):
+
+    mydir = TestBase.compute_mydir(__file__)
+
+    eBroadcastBitStopProgressThread = (1 << 0)
+
+    def setUp(self):
+        TestBase.setUp(self)
+        self.progress_events = []
+
+    def fetch_events(self):
+        event = lldb.SBEvent()
+
+        done = False
+        while not done:
+            if self.listener.WaitForEvent(1, event):
+                event_mask = event.GetType();
+                if event.BroadcasterMatchesRef(self.test_broadcaster):
+                    if event_mask & self.eBroadcastBitStopProgressThread:
+                        done = True;
+                elif event.BroadcasterMatchesRef(self.progress_broadcaster):
+                    ret_args = lldb.SBDebugger().GetProgressFromEvent(event);
```

Comment:
should `GetProgressFromEvent` be a static function?

Reply:
It is static internally. To unpack the event data, we don't need to have access to a specific instance of the debugger.

Reply:
Seems like we're starting to accumulate a few of these functions. Can we put this in a helper function upstream that's generic enough to be used by this tests as well as the two upstream ones?

Reply:
@JDevlieghere I'll do that in a follow-up patch